lib/async_hooks.js

```diff
@@ -326,7 +326,14 @@ function emitInitS(asyncId, type, triggerAsyncId, resource) {
   if (!Number.isSafeInteger(triggerAsyncId) || triggerAsyncId < 0)
     throw new RangeError('triggerAsyncId must be an unsigned integer');
 
-  init(asyncId, type, triggerAsyncId, resource);
+  // Remove prepear for a fatal exception in case an error occurred
+  var didThrow = true;
+  try {
+    init(asyncId, type, triggerAsyncId, resource);
+    didThrow = false;
+  } finally {
+    if (didThrow) clearFatalExceptionHandlers();
```

Comment:
I'm not sure it's correct to move this line from the `N` variant to the `S` variant?